### PR TITLE
test/1556/orderTable

### PR DESCRIPTION
**Pull request by mrjnmzk**
## Description

rewrite test to React Testing Library

### Checklist

- [x] 🔽 My branch is up-to-date with "development" branch
- [x] ✅All tests passed locally
- [x] ✨My changes working with up-to-date admin and back-end part locally, like charm
- [x] 🔗 Link pull request to issue

### CODE REVIEW

**Review by maarekua — APPROVED** [review]

File: src/containers/orders/order/order-table/tests/order-table.test.js [hunk 1]
```diff
@@ -1,35 +1,164 @@
...
+          },
+          {
+            value: 37,
+            currency: 'USD'
+          }
+        ],
+        size: { _id: 'test', name: 'S' }
+      }
+    }
+  ],
```
Comment (byavdoshniak) [inline src/containers/orders/order/order-table/tests/order-table.test.js:68]: Maybe we should remove huge constants to test.variables.js and import them here.

File: src/containers/orders/order/order-table/tests/order-table.test.js [hunk 2]
```diff
@@ -1,35 +1,164 @@
...
+    act(() => {
+      fireEvent.click(confirmButton);
+    });
+    expect(modalProps.onAction).toHaveBeenCalledWith(true);
+  });
+
+  it('call onAction with false', () => {
+    const cancelButton = screen.queryByText('common.buttons.cancel');
+
+    expect(cancelButton).toBeInTheDocument();
```
Comment (byavdoshniak) [inline src/containers/orders/order/order-table/tests/order-table.test.js:148]: This expects don't do anything, cause if fireEvent is triggered, then it's obvious that we have fireEvent.target in the document.
Same for all test cases besides first

**Review by byavdoshniak — CHANGES_REQUESTED** [review]

File: src/containers/orders/order/order-table/tests/order-table.test.js (hunk 1 above)
Comment (mrjnmzk) [reply src/containers/orders/order/order-table/tests/order-table.test.js]: hi. i removed some constants. which i dont use

Comment (mrjnmzk) [reply src/containers/orders/order/order-table/tests/order-table.test.js]: if i import a constant the test can't reach them

**Review by byavdoshniak — DISMISSED** [review]

File: src/containers/orders/order/order-table/tests/order-table.test.js [hunk 3]
```diff
@@ -1,35 +1,155 @@
...
 
 jest.mock('../order-table.styles', () => ({
   useStyles: () => ({})
 }));
+jest.mock('../../../../../components/modal/modal.styles', () => ({ useStyles: () => ({}) }));
+jest.mock('../../../cart/cart-item/cart-item.styles', () => ({ useStyles: () => ({}) }));
 jest.mock('react-redux');
-
-const mockCartOperations = { removeFromCart: jest.fn() };
+jest.mock('@apollo/client', () => ({
```
Comment (maarekua) [inline src/containers/orders/order/order-table/tests/order-table.test.js:18]: https://www.apollographql.com/docs/react/development-testing/testing/

**Review by maarekua — CHANGES_REQUESTED** [review]

File: src/containers/orders/order/order-table/tests/order-table.test.js [hunk 4]
```diff
@@ -1,35 +1,143 @@
...
   useSelector.mockImplementation(() => ({
-    language: lang
+    language: lang,
+    currency: 0
   }));
 };
-let wrapper;
+
+jest.mock('react-redux');
+jest.mock('react-i18next', () => ({
```
Comment (maarekua) [inline src/containers/orders/order/order-table/tests/order-table.test.js:27]: We have global mock for i19next.

File: src/containers/orders/order/order-table/tests/order-table.test.js [hunk 5]
```diff
@@ -1,35 +1,134 @@
...
+};
+
+describe('test <OrderTable /> component', () => {
+  it('should render <OrderTable />', () => {
     testUseSelector(0);
-    wrapper = shallow(<OrderTable {...props} cartOperations={mockCartOperations} />);
-    expect(wrapper).toMatchSnapshot();
+    const { getByText } = render(
+      <BrowserRouter>
+        <MockedProvider>
```
Comment (maarekua) [inline src/containers/orders/order/order-table/tests/order-table.test.js:86]: Cant find using of the Mocked provider, and it configuration.

**Review by maarekua — CHANGES_REQUESTED** [review]

File: src/containers/orders/order/order-table/tests/order-table.test.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (mrjnmzk) [reply src/containers/orders/order/order-table/tests/order-table.test.js]: fixed

File: src/containers/orders/order/order-table/tests/order-table.test.js [hunk 6]
```diff
@@ -1,35 +1,143 @@
...
 
-const mockCartOperations = { removeFromCart: jest.fn() };
-
 const dispatch = jest.fn();
 useDispatch.mockImplementation(() => dispatch);
+
 const testUseSelector = (lang) => {
   useSelector.mockImplementation(() => ({
-    language: lang
+    language: lang,
```
[the remark this answers is not in the document]
Comment (mrjnmzk) [reply src/containers/orders/order/order-table/tests/order-table.test.js]: fixed

File: src/containers/orders/order/order-table/tests/order-table.test.js (hunk 5 above)
Comment (mrjnmzk) [reply src/containers/orders/order/order-table/tests/order-table.test.js]: fixed

**Review by maarekua — APPROVED** [review]

**Review by boris-giga — APPROVED** [review]

**Pull request merged**